os/board/rtl8721csm/src/component/common/file_system/ftl/ftl.c

```diff
@@ -1204,11 +1196,15 @@ uint32_t ftl_non_secure_save_to_storage(void *pdata_tmp, uint16_t offset, uint16
 
 uint32_t ftl_save_to_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
 {
+	device_mutex_lock(RT_DEV_LOCK_FLASH);
 #if defined(CONFIG_AMEBAD_TRUSTZONE) && (FTL_SECURE == 1)
-	return ftl_secure_save_to_storage(pdata_tmp, offset, size);
+	int ret = ftl_secure_save_to_storage(pdata_tmp, offset, size);
 #else
-	return ftl_non_secure_save_to_storage(pdata_tmp, offset, size);
+	int ret = ftl_non_secure_save_to_storage(pdata_tmp, offset, size);
 #endif
+	device_mutex_unlock(RT_DEV_LOCK_FLASH);
+	return ret;

```

Comment:
The function return type is `unit32_t` . But the `ret` is `int`. It has type mismatch. This will cause a SVACE defect.